### DISCUSSION TITLE
Users authenticated via an IdP don't show up in the Users list

### DISCUSSION THREAD

**Discussion by neverstew** [zitadel/zitadel#6074] [Q&A]
Let me describe a scenario I was testing this evening. It didn't go quite as I expected...

There are two Orgs: Big Tech and Little Tech.
Each has a project called "default" with a web application called "web".
Little Tech has configured Big Tech as an Identity Provider, using the OIDC config it exposes. This is the only way to sign into Little Tech projects.

"John" signs up via the Little Tech web application and manages to get all the way through verification etc. and makes it to the Little Tech application callback URL successfully.
An organisation owner, "Steve", at Little Tech logs into zitadel to check the Users list.

Steve expects to see John in the Users list, but doesn't.

Is this something that you'd also expect, or is this something that isn't supposed to happen? In the case where it's not supposed to happen, what mechanisms are there to see "any user that has logged in to my application, regardless of what mechanism they used to log in"?

**Comment by hifabienne** [comment]
Hei @neverstew 
I am not sure if I completely understand your use case.
First, what is your use case to add ZITADEL as an identity provider within ZITADEL?
With the delegated access management we have implemented, you are also able to give users outside your organization authorizations to you projects.

Generally when using an external identity provider, the user also has to exist in ZITADEL. E.g when using Google as Identity Provider, a user in ZITADEL exists that has Google linked.
I haven't actually tested using my own ZITADEL instance as idp / only using another ZITADEL instance, because I don't see the use case. Would have to check that.

**Comment by neverstew** [comment]
Hi @hifabienne ,

The fact that one zitadel application is configured as the IdP for another on the same instance is somewhat arbitrary, it was just an idea I was testing. The main thing was that users, signing in via an IdP, don't show up in the users list.

Is that expected? If not, then maybe it is something to do with the specific scenario, but maybe not. I'm just trying to check my expectations first.

**Reply by hifabienne** [reply]
Generally is not expected, as each "external user" also has a local account in ZITADEL.

**Reply by neverstew** [reply]
I was expecting each "external user" to also have a local account in zitadel as well, so at least we're on the same page there.

So the question remains, why didn't this happen when the "external IdP" happens to be a zitadel application on the same instance...

**Reply by hifabienne** [reply]
Does the account in the big tech organization now have an entry on the external authentication list?

**Reply by neverstew** [reply]
Good question - I'm away from my machine at the moment so I'll check later. If you think of any other things to check before then feel free to send them over!

**Reply by neverstew** [reply]
Right I found some time and started from scratch to make sure it all made sense and I could present it in the simplest way possible...

# Context
There are two orgs, as before, here's more about them...

## Big Tech
The default org for the instance.
Has one application, using POST OIDC configuration.
Registration on.
No external IdP.

## Little Tech
Has one application, using PKCE.
Registration off.
One external IdP, set to OIDC with Big Tech's credentials.

# Users
## Emma
Manually added as a user in Big Tech.
Is an Organisation User Manager at Little Tech.

## Kate
A regular user of Little Tech's web app.
She registered by visiting Little Tech's web app, getting automatically directed to the Big Tech login, registering and then signing in.
She is a user in the Big Tech organisation.

# Problem
Emma cannot see Kate. She's not in any list on the Little Tech console.

**Reply by neverstew** [reply]
Found some more time to approach this from another simpler angle, using domain discovery rather than Identity Providers...

# Context
## Big Tech
The default org for the instance.
Has one application, using PKCE OIDC configuration.
Registration on.

## Little Tech
Has one application, using PKCE.
Registration off.
Domain discovery on.

# Users
## Emma
Registered herself via the Little Tech web app.
Is a user in the Big Tech org.
Is a User Manager in the Little Tech org.

## Kate
Registered herself via the Little Tech web app.
Is a user in the Big Tech org.

# Problem
Emma's not in either Users or Authorizations on the Little Tech console.
Kate's not in either Users or Authorizations on the Little Tech console.

Emma has no way of seeing Kate.

**Reply by hifabienne** [reply]
Can you please describe your use case? What do you want to achieve?
For me it sounds like your B2B scenario? But I am not sure what you want to do.
https://zitadel.com/docs/guides/solution-scenarios/b2b

Is it about having one organization for the company which manages the projects and applications, and then having organizations for the customers?

**Reply by neverstew** [reply]
Yes, it's definitely a B2B scenario. Here are some requirements:
1. one, centralised identity provider
2. many organisations that can each be self-managed by other businesses
3. each organisation should be able to see the users that have logged into their system 

Number three is not working, as described above!

**Reply by hifabienne** [reply]
Thanks for clarifying.
Generally some insight. ZITADEL is organized around organizations. Each user belongs to one organization, but can have permissions in multiple. The organization that manages the users, is the organization where the users should belong.
In your case I guess the users should actually exist in the Little Tech Organization and not in the Big Tech organization.

You can either achieve this by
1. An administrator creates the users directly in the organization where they should belong or
2. Implementing your own registration UI, make sure to specify the organization where you want to manage the ressources. When using the management API, the requests are performed on the organization where the requesting user belongs. If you want to send the request for another organization please send the org header. https://zitadel.com/docs/apis/introduction#management
3. Self registration through ZITADEL registration UI. When the users self register in the ui it need so be clear to which organization, you can achieve this by sending the organization scope when redirecting from the application to the ZITADEL login urn:zitadel:iam:org:id:{id}: https://zitadel.com/docs/apis/openidoauth/scopes#reserved-scopes

The things mentioned above should all result in seeing the users in the users list of the organization they belong.

What at the moment is missing in ZITADEL is a view where you can see which users has authenticated to which application.

**Reply by neverstew** [reply]
Yes, that all sounds like a very sensible approach for a lot of scenarios though not the the one I want to take. 

There is currently no way for organisations to see users that have accessed their applications via IdP or domain discovery, so how are they supposed to keep track of their users?
